### PR TITLE
chore: refractor to use RxExtensions

### LINKED ISSUE #1696: refractor to use RxExtensions
**Describe the chore**
Refractor entire project to use RxExtension with 
`fun <T> Single<T>.with(): Single<T> = subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())`
to write
`  compositeDisposable.add(sponsorService.fetchSponsorsWithEvent(id)
            .with()`
 instead of writing
`   compositeDisposable.add(sponsorService.fetchSponsorsWithEvent(id)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe({})
`     
**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1696 

Changes: Refractor entire project to use RxExtensions

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal @nikit19 review please

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt [hunk 1]
```diff
@@ -30,8 +29,7 @@ class AboutEventViewModel(private val eventService: EventService, private val re
             return
         }
         compositeDisposable.add(eventService.getEvent(id)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+        .with()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: `with` makes no sense

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: @iamareebjamal should I change the name of extension?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: Yes

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: any suggestions? @iamareebjamal

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: with -> subscribeWith will that be okay or something else

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: `withDefaultSchedulers`

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: @iamareebjamal updated

**Comment by iamareebjamal** [comment]
Fix build, add RxKotlin and change `compositeDisposable.add` to `compositeDisposable +=`

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated

File: app/build.gradle [hunk 2]
```diff
@@ -153,6 +153,7 @@ dependencies {
     implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
     implementation 'io.reactivex.rxjava2:rxjava:2.2.8'
     implementation 'com.squareup.retrofit2:adapter-rxjava2:2.4.0'
+    implementation 'io.reactivex.rxjava2:rxkotlin:2.3.0'
```
Comment (iamareebjamal) [inline app/build.gradle]: Should be above RxJava

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt [hunk 3]
```diff
@@ -29,9 +29,8 @@ class AboutEventViewModel(private val eventService: EventService, private val re
             mutableError.value = Resource().getString(R.string.error_fetching_event_message)
             return
         }
-        compositeDisposable.add(eventService.getEvent(id)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+        compositeDisposable += (eventService.getEvent(id)
+        .withDefaultSchedulers()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: Wrong indentation

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt [hunk 4]
```diff
@@ -29,9 +29,8 @@ class AboutEventViewModel(private val eventService: EventService, private val re
             mutableError.value = Resource().getString(R.string.error_fetching_event_message)
             return
         }
-        compositeDisposable.add(eventService.getEvent(id)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+        compositeDisposable += (eventService.getEvent(id)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventViewModel.kt]: Remove leading bracket

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal updated

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 5]
```diff
@@ -132,9 +132,8 @@ class AttendeeViewModel(
         }
         mutableQtyList.value = qty
 
-        compositeDisposable.add(ticketService.getTicketPriceWithIds(ticketIds)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
+        compositeDisposable += ticketService.getTicketPriceWithIds(ticketIds)
+                .withDefaultSchedulers()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: indentation

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Updated.Anything else ?

**Pull request merged**